pkg/trace/agent/agent.go

```diff
@@ -295,19 +295,27 @@ func (a *Agent) processChunk(chunk *pb.TraceChunk, p *api.Payload, now time.Time
 	}
 
 	numEvents, keep, filteredChunk := a.sample(now, ts, pt)
-	if !keep && numEvents == 0 {
-		// the trace was dropped and no analyzed span were kept
+
+	// If sampling decided to drop this chunk, then there are three possibilities:
```

Comment:
nit: I'd suggest moving these details into the respective if-statements so it isn't quite so verbose. You've already started doing that anyway, so you can just flesh it out there rather than saying things twice.